### PR TITLE
Change the placement of the feature flag check for Ad Blocking Recovery settings

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue:

- #7179

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by nfmohit — CHANGES_REQUESTED** [review]
Thank you for the work here, @tofumatt, absolutely accurate!

I'm being very nitpicky here. I understand updating stories/tests wasn't mentioned in the IB, however, test suites for both the `AdBlockingRecoveryCTA` and `AdBlockingRecoveryToggle` components include tests to verify if the `adBlockerDetection` feature flag is enabled.

`AdBlockingRecoveryCTA`:
https://github.com/google/site-kit-wp/blob/9fa3d850478fb154a9baa1bf6ca6087d640c2c16/assets/js/modules/adsense/components/common/AdBlockingRecoveryCTA.test.js#L37-L43

`AdBlockingRecoveryToggle`:
https://github.com/google/site-kit-wp/blob/9fa3d850478fb154a9baa1bf6ca6087d640c2c16/assets/js/modules/adsense/components/common/AdBlockingRecoveryToggle.test.js#L45

Also, all the other tests within the above test suites set the `adBlockerDetection` feature flag to be enabled.

The tests still work flawlessly with them existing there. Now that the feature flag checks have been removed from the components within, do you think these checks are redundant and should be removed from the tests?

Also, the Storybook stories for `AdBlockingRecoveryCTA` and `AdBlockingRecoveryToggle` components set the `adBlockerDetection` feature flag to be enabled. For example:

https://github.com/google/site-kit-wp/blob/9fa3d850478fb154a9baa1bf6ca6087d640c2c16/assets/js/modules/adsense/components/common/AdBlockingRecoveryCTA.stories.js#L44-L46

Do you think these should also be removed?

**Comment by tofumatt** [comment]
Nitpicks are okay, but also these are hardly nitpicks, great catches! 🙏🏻 

> The tests still work flawlessly with them existing there. Now that the feature flag checks have been removed from the components within, do you think these checks are redundant and should be removed from the tests?

Yes, they should be removed. Thank you for finding/noticing this! These `it.each` make for very unreadable tests in my view, and this is a prime example of why: it was _totally_ unobvious to me that the uncommented/named variable there was adblocking detection unless I read through dozens of line of code 😆.

For future reference to any devs reading this, arrays of something like:

```js
{
  testName: 'Adsense site status is not ready',
  accountStatus: ACCOUNT_STATUS_READY,
  siteStatus: SITE_STATUS_ADDED,
  recoverySetupStatus: '',
  adBlockerDetectionFeatureFlag: true,
},
```

over

```js
[
  'Adsense site status is not ready',
  ACCOUNT_STATUS_READY,
  SITE_STATUS_ADDED,
  '',
  true,
],

for `it.each` would massively improve readability… though really we just shouldn't be using `it.each` to my mind as it's a tool for writing tests quick, and tests should be slow to write and quick to read 😉.

**Comment by nfmohit** [comment]
I absolutely agree, thank you for the example!

**Review by nfmohit — APPROVED** [review]
LGTM, thank you!

File: assets/js/modules/adsense/components/settings/SettingsForm.js [hunk 1]
```diff
@@ -125,9 +128,12 @@ export default function SettingsForm() {
 
 			<AutoAdExclusionSwitches />
 
-			<AdBlockingRecoveryCTA />
-
-			<AdBlockingRecoveryToggle />
+			{ adBlockerDetectionEnabled && (
+				<Fragment>
+					<AdBlockingRecoveryCTA />
```
Comment (aaemnnosttv) [inline assets/js/modules/adsense/components/settings/SettingsForm.js:133]: This component is also rendered in the `SettingsView` which is still being rendered unconditionally.

https://github.com/google/site-kit-wp/blob/94ecff3e5ac9587fed96afde2989f3849f512df8/assets/js/modules/adsense/components/settings/SettingsView.js#L225-L228 

Apologies, I missed this in the IB too.

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @tofumatt ! This is almost there, just one more thing which is needed.

**Comment by aaemnnosttv** [comment]
A few more things I noticed when looking at the `SettingsView`, only one of which is relevant here: the ABR setup status is only conditional based on its value, not the feature flag so this should also be addressed here

https://github.com/google/site-kit-wp/blob/94ecff3e5ac9587fed96afde2989f3849f512df8/assets/js/modules/adsense/components/settings/SettingsView.js#L182

Secondarily, I don't think these messages match the descriptions
https://github.com/google/site-kit-wp/blob/94ecff3e5ac9587fed96afde2989f3849f512df8/assets/js/modules/adsense/components/settings/SettingsView.js#L188-L204
I noticed this when the setup was completed and both toggles were on, but the view was showing `Ad blocking recovery tag is not placed`.

The tag is placed for both of these statuses – it's the blank string where they're not placed (this would probably be better as a named value as well. We also have an enum value for the statuses now as well but that's a minor detail. This one should be revisited and corrected separately.

I'll push a fix for the additional guards that were missed and send it back to you @tofumatt for MR 👍 

cc: @kuasha420 ☝️

**Review by aaemnnosttv — APPROVED** [review]

**Review by tofumatt — COMMENTED** [review]
Looks good, thanks for catching that one @aaemnnosttv! 👍🏻

**Pull request merged**